app-cpp/app355.cpp

```diff
@@ -329,14 +329,17 @@ void SatOrbit::SortOrbitalVectorMulti(std::vector<OrbitalData>& ioOrbitalVector)
 
     // calculate the number of TLEs per thread
     const std::size_t orbitPerThread = orbitVectorSize / mNumThreads;
-    // Note: no need for mutex as threads do not contend for shared elements during sort
     IteratorPairVector subListVector{};
 
+    // Lock the orbit mutex as both calculate and sort could be multithreaded
+    auto& [mutex, outputVector] = mOrbitalVector;
+    std::lock_guard<std::mutex> lock(mutex);
```

Comment:
hmmm... this doesn't work at all.
the vector being modified is `ioOrbitalVector`...
but here you try to lock `mOrbitalVector`.
the code here has no idea whether those are the same vector!

also, what about "calculate"?
wouldn't it also want to modify `mOrbitalVector`?
i think locking is needed there also!